### DISCUSSION TITLE
Why does the nuget package include two winmd versions?

### DISCUSSION THREAD

**Discussion by tristanlabelle** [microsoft/WindowsAppSDK#4079] [Q&A] [answered]
The Windows App SDK nuget package includes `10.0.17763.0` and `10.0.18362.0` versions of its winmds. Why is that?

If my app's minimum requirements are above `10.0.18362.0`, am I better to use that version? What difference does it make?

I'm sure this has been answered somewhere or is documented but my searches couldn't find it.

Thanks!

**Comment by DarranRowe** [comment, marked as answer]
Currently, it is due to Microsoft.UI.Input.Interop.PenDeviceInterop.

I can't really comment on how they split the .winmd files. However, Microsoft.UI.winmd contains the definition for the PenDeviceInterop runtime class and the IPenDeviceInteropStatics interface that provides the static member function definitions for the runtime class. If you look at the [PenDeviceInterop](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.interop.pendeviceinterop?view=windows-app-sdk-1.4) documentation, you will notice that this contains a single static member function named FromPointerPoint. This static member function returns an instance of [Windows.Devices.Input.PenDevice](https://learn.microsoft.com/en-us/uwp/api/windows.devices.input.pendevice?view=winrt-22621). This is where the issue is. The documentation for PenDevice states that it is available starting with Windows 10 build 18362. If you used the 18362 targeted version of the Windows App SDK with the 17763 version of the Windows metadata (either the Windows SDK or the SDK contract references) then it would be missing a reference.
The version you should use is thus tied to the version of the Windows .winmd files you are using. Visual Studio should take care of things automatically if you are using Visual Studio as your build system. If you are using something else then you should use the 17763 version if you are using the 17763 version of the Windows .winmd files, obtained from the Windows SDK, the contract reference NuGet package or even Windows itself from %SystemRoot%\System32\WinMetadata. You should use the 18362 version otherwise.
It isn't the end of the world if you use the 17763 version with newer versions of Windows. You would just lose the PenDeviceInterop runtime class.

**Reply by tristanlabelle** [reply]
Ah, makes sense! Thanks for clarifying.
